packages/yoastseo/src/languageProcessing/researches/getLinkStatistics.js

```diff
@@ -17,33 +17,39 @@ let functionWords = [];
 /**
  * Checks whether the link is pointing at itself.
  *
+ * @param {string} slug      	 	  The slug.
  * @param {string} anchor      	 	  The link anchor.
  * @param {string} siteUrlOrDomain    The site URL or domain of the paper.
  *
  * @returns {boolean} Whether the anchor is pointing at itself.
  */
-const linkToSelf = function( anchor, siteUrlOrDomain ) {
+const linkToSelf = function( slug, anchor, siteUrlOrDomain ) {
 	const anchorLink = urlHelper.getFromAnchorTag( anchor );
-
 	// Relative fragment links always point to the page itself.
 	if ( urlHelper.isRelativeFragmentURL( anchorLink ) ) {
 		return true;
 	}
-
-	return urlHelper.areEqual( anchorLink, siteUrlOrDomain );
+	// If siteUrlOrDomain is a domain, create URLs out of it in order to compare them with the anchor URL.
+	if ( ! new RegExp( "^(https://|http://)" ).test( siteUrlOrDomain ) && siteUrlOrDomain !== "" ) {
+		siteUrlOrDomain = [ "http://" + siteUrlOrDomain + "/" + slug, "https://" + siteUrlOrDomain + "/" + slug ];
```

Comment:
This will not work in all cases, unfortunately 😞 .

The slug contains the last part of the URL, after the last slash. We would be missing anything between the domain (`example.com`) and the slug.

For example, the _goldfish_ product's permalink on my test site would be 
```
https://hansjovis-fish-supplies.com/products/goldfish
```

The slug would be `goldfish` and the primary domain would be `hansjovis-fish-supplies`.

The `/products/` path in the middle would be missing.